src/structured-data/structured-data-tags.js

```diff
@@ -218,9 +218,15 @@ function generateBreadcrumbListData(pageType = "", publisherConfig = {}, data =
     return sectionCrumbsDataList;
   }
 
+  function getCustomPageCrumbs(collection = {}) {
+    const crumbsDataList = [{ url: collection.pagePath, name: collection.name}];
+    return addCrumb(crumbsDataList, collection);
+  }
+
   switch (pageType) {
     case "section-page": breadcrumbsDataList = breadcrumbsDataList.concat(getSectionPageCrumbs(data.section)); break;
     case "story-page": breadcrumbsDataList = breadcrumbsDataList.concat(getStoryPageCrumbs(data.story)); break;
+    case "custom-page": breadcrumbsDataList = breadcrumbsDataList.concat(getCustomPageCrumbs(data.collection)); break;
```

Comment:
Please dont add a pageType to this. If there is `data.breadcrumbs` please use that. Basically something like this.
```js
  if(data.breadcrumbs && data.breadcrumbs.length > 0) {
    breadcrumbsDataList = breadcrumbsDataList.concat(getCustomPageCrumbs(data.breadcrumbs))
  }
  switch (pageType) {
    case "section-page": breadcrumbsDataList = breadcrumbsDataList.concat(getSectionPageCrumbs(data.section)); break;
    case "story-page": breadcrumbsDataList = breadcrumbsDataList.concat(getStoryPageCrumbs(data.story)); break;
  }
```

Reply:
@anbusurendhar-vikatan Can you check this also?

Reply:
Please do not use custom-page. You will not have any page called as "custom-page".

Reply:
Removed custom-page in switch case. Also written test for the modified code. Can you review it once?